Sort milestones by closest due date

smeijer opened the pull request:
The default sorting of milestones in github is pretty frustrating. Milestones are used to manage releases; where as the closest one is often more important than the last updated one.

This PR fixes the default sorting, so they are sorted by closest due date, instead of recently updated.

sindresorhus commented:
Clicking the "Milestones" button in https://github.com/sindresorhus/got/issues?q=is:open gives me a 404.

sindresorhus commented:
Can you add it to https://github.com/sindresorhus/refined-github#highlights ? No screenshot needed.

bfred-it commented:
jQuery returns an array even if it's not found so `if array` will always be true. Refer to #454 for the suggested implementation.

bfred-it commented on extension/content.js:
```diff
@@ -430,6 +430,13 @@ function addTitleToEmojis() {
 }
 
+function sortMileStonesByClosestDueDate() {
+	const button = $('.subnav-item[href$="milestones"');
```
Missing `]` in selector

bfred-it commented on extension/content.js:
```diff
@@ -430,6 +430,13 @@ function addTitleToEmojis() {
 
+function sortMileStonesByClosestDueDate() {
+	const button = $('.subnav-item[href$="milestones"');
+	if (button) {
```
jQuery returns an array even if it's not found so if array will always be true. Refer to #454 for the suggested implementation.

hkdobrev commented:
@smeijer Do you think you'd be able to pick this up? Thank you!

smeijer commented:
Sure; I'll fix this asap.

smeijer commented:
@zackschuster , 
- [x] rebased on master
- [x] implemented a native js solution as requested by @bfred-it : https://github.com/sindresorhus/refined-github/pull/457#discussion_r119766742
- [x] added to highlights as requested by @sindresorhus : https://github.com/sindresorhus/refined-github/pull/457#issuecomment-305576850

sindresorhus commented:
It doesn't work if you click the `Milestone` button in https://github.com/chalk/chalk/labels and https://github.com/chalk/chalk/milestone/1

smeijer commented:
@sindresorhus , I added some page detection's to fix this. It does work for me now, but I have no idea if this is fits in the projects standards. Please take this PR over or close it if it still needs some work.

sindresorhus commented on src/content.js:
```diff
@@ -594,6 +602,11 @@ function init(options) {
+			}
+
+			if (pageDetect.isMilestoneList() || pageDetect.isLabelList() || pageDetect.isLabel()) {
+				sortMileStonesByClosestDueDate();
```
Better to have one if statement with all of these `pageDetect`s than spreading `sortMileStonesByClosestDueDate()` in multiple different ones.

sindresorhus commented:
Can you fix the merge conflict?

bfred-it commented:
To make it truly default, it should touch all the links to **Milestones** that are not explicitly in a different order. These include the Open and Closed links and those added by #366 

<img width="224" alt="screen shot 2017-07-05 at 22 44 12" src="https://user-images.githubusercontent.com/1402241/27869798-9d101f4c-61d3-11e7-930d-2864b7bbbef7.png">

bfred-it commented:
Should we get this in?

sindresorhus commented on readme.md:
```diff
@@ -135,6 +135,22 @@ GitHub Enterprise is also supported by [authorizing your own domain in the optio
+
+![](media/screenshot-dashboard.png)
+
+### Repo
```
This merge conflict seems to have been resolved incorrectly.

bfred-it replied:
Good catch. I had only ran `git rebase origin/master`
